fighters/duckhunt/src/opff.rs

```diff
@@ -4,44 +4,130 @@ use super::*;
 use globals::*;
 
  
-unsafe fn duck_jump_cancel(boma: &mut BattleObjectModuleAccessor, status_kind: i32, cat1: i32, frame: f32) {
-    if status_kind == *FIGHTER_DUCKHUNT_STATUS_KIND_SPECIAL_HI_FLY {
-        if frame > 20.0 {
-            if boma.is_cat_flag(Cat1::SpecialHi) {
-                StatusModule::change_status_request_from_script(boma, *FIGHTER_DUCKHUNT_STATUS_KIND_SPECIAL_HI_END, true);
-            }
-        }
+unsafe fn duck_jump_cancel(fighter: &mut L2CFighterCommon) {
+    if fighter.is_status(*FIGHTER_DUCKHUNT_STATUS_KIND_SPECIAL_HI_FLY)
+    && fighter.motion_frame() > 20.0
+    && fighter.is_cat_flag(Cat1::SpecialHi) {
+        fighter.change_status_req(*FIGHTER_DUCKHUNT_STATUS_KIND_SPECIAL_HI_END, true);
     }
 }
 
-pub unsafe fn moveset(boma: &mut BattleObjectModuleAccessor, id: usize, cat: [i32 ; 4], status_kind: i32, situation_kind: i32, motion_kind: u64, stick_x: f32, stick_y: f32, facing: f32, frame: f32) {
-    duck_jump_cancel(boma, status_kind, cat[0], frame);
+unsafe fn frame_data(fighter: &mut L2CFighterCommon) {
+    if fighter.is_status(*FIGHTER_STATUS_KIND_SPECIAL_N) {
```

Comment:
i know i changed the calls for these rq and gave them to you, but they could be very slightly improved with a match. not really important though

Reply:
https://doc.rust-lang.org/1.6.0/book/patterns.html#guards
I think it would become more verbose. I feel like leaving at two IFs is more readable personally. I would switch to a match with 3 values or more.